torchaudio_contrib/augmentation.py

```diff
@@ -0,0 +1,151 @@
+    """
+    Stretch stft in time without modifying pitch for a given rate.
+
+    Args:
+
+        rate (float): rate to speed up or slow down by. Defaults to 1.
+        hop_len (int): Number audio of frames between STFT columns.
+            Defaults to 512.
+        num_bins (int, optional): number of filter banks from stft.
+            Defaults to 1025.
+    """
+
+    def __init__(self, rate=1., hop_len=512, num_bins=1025):
+
+        super(StretchSpecTime, self).__init__()
+
+        self.rate = rate
+        phi_advance = torch.linspace(
+            0, math.pi * hop_len, num_bins)[..., None]
+
+        self.register_buffer('phi_advance', phi_advance)
+
+    def forward(self, spect, rate=None):
+        if rate is None:
+            rate = self.rate
+        return phase_vocoder(spect, rate, self.phi_advance)
+
+    def __repr__(self):
+        param_str = '(rate={})'.format(self.rate)
+        return self.__class__.__name__ + param_str
+
+
+class GaussianNoise(_ModuleNoStateBuffers):
+    """
+    Apply gaussian noise to a spectrogram.
+
+    Args:
+        scale (float or tensor): standard deviation of the distribution.
+    """
+
+    def __init__(self, scale=1.):
+        super(GaussianNoise, self).__init__()
+        self.scale = torch.as_tensor(scale)
+        self.loc = torch.tensor(0.)
+
+    def forward(self, spect, scale=None):
+        if scale == None:
+            scale = self.scale
+        else:
+            scale = torch.as_tensor(scale)
+
+        with torch.no_grad():
+
+            noise = torch.normal(self.loc.expand(spect.shape), scale.expand(spect.shape)) 
+            return spect, noise
+
+    def __repr__(self):
+        param_str = '(scale={})'.format(self.scale)
+        return self.__class__.__name__ + param_str
+
+
+def apply_masking(spect, max_value, mask_value, axis):
+
+    assert axis in [2,3]
+
+    v = torch.rand(spect.shape[:2])*max_value
+    v_0 = torch.rand(spect.shape[:2])*(spect.size(axis) - v)
+
+    mask_start = (v_0.long()).unsqueeze(-1)
+    mask_end = (v_0.long() + v.long()).unsqueeze(-1)
+
+    mask = torch.arange(0, spect.size(axis)).repeat(spect.size(0), spect.size(1),1)
+    
+    spect = spect.transpose(2, axis)
+
+    # per (batch, channel) mask
+    spect[(mask >= mask_start) & (mask < mask_end)] = mask_value
+    spect = spect.transpose(2, axis)
+    
+    return spect
+
+
+def apply_masking_batch(spect, max_value, mask_value, axis):
+
+    v = torch.rand(1)*max_value
+    v_0 = torch.rand(1)*(spect.size(axis) - v)
+
+    mask_start = (v_0.long()).squeeze()
+    mask_end = (v_0.long() + v.long()).squeeze()
+
+    if axis == 2:
+        spect[:,:,mask_start:mask_end] = mask_value
+    elif axis == 3:
+        spect[:,:,:,mask_start:mask_end] = mask_value
+    else:
+        raise ValueError('Only Frequency and Time masking is supported')
+
+    return spect
+
+
+class _AxisMasking(_ModuleNoStateBuffers):
+
+    def __init__(self, max_value, axis, same_batch):
```

Comment:
`same_batch` --> something more intuitive? I couldn't get what it means from the name.

Reply:
How about `along_batch`? `same_along_batch` might be too verbose..

Reply:
or, `mask_per_sample`? `per_sample`? hm, not easy :o `batchwise_mask`? ...

Reply:
On things like this, prob we can simply add a comment saying the name doesn't sound optimal and proceed for the moment.